app/forms/provider_interface/offer_condition_fields.rb

```diff
@@ -0,0 +1,9 @@
+module ProviderInterface
+  class OfferConditionFields
+    include ActiveModel::Model
+
+    attr_accessor :id, :text
+
+    validates :text, length: { maximum: 255, too_long: ->(c, _) { "Condition #{c.id + 1} must be %{count} characters or fewer" } }
```

Comment:
Consider moving the validation error in the make offer locale file for consistency

Reply:
I had a look into this, but it doesn't seem to work.

It is possible to pass attributes through to locales by doing e.g. `validates :text, length: { maximum: 255,  attribute: 'whatever' }`, but we cannot refer to the ID of the condition in those, because adding validations happens at the class level, rather than at the instance level.

We rely on the too_long method passing through the instance so we can refer to the id in the string. We can revisit this at a later date